llvm/utils/TableGen/Common/GlobalISel/GlobalISelMatchTable.cpp

```diff
@@ -237,9 +237,10 @@ MatchTableRecord MatchTable::NamedValue(unsigned NumBytes, StringRef Namespace,
 
 MatchTableRecord MatchTable::IntValue(unsigned NumBytes, int64_t IntValue) {
   assert(isUIntN(NumBytes * 8, IntValue) || isIntN(NumBytes * 8, IntValue));
-  auto Str = llvm::to_string(IntValue);
-  if (NumBytes == 1 && IntValue < 0)
-    Str = "uint8_t(" + Str + ")";
+  uint64_t UIntValue = IntValue;
+  if (NumBytes < 8)
+    UIntValue &= (UINT64_C(1) << NumBytes * 8) - 1;
+  auto Str = llvm::to_string(UIntValue);
```

Comment:
```suggestion
  std::string Str = llvm::to_string(UIntValue);
```
We probably should be directly writing this string to the stream instead of creating the temporary

Reply:
> We probably should be directly writing this string to the stream instead of creating the temporary

I can address that in a future PR, together with the following TODO comment of emitting the bytes directly instead of via `GIMT_Encode` macros.